generic/tests/dd_test.py

```diff
@@ -51,73 +52,101 @@ def _get_file(filename, select):
                 err = ("Incorrect cfg: dd_select out of the range (disks=%s,"
                        " select=%s)" % (disks, select))
                 logging.error(err)
-                raise error.TestError(err)
+                test.error(err)
+
+    class CmdFormat(object):
+        def __init__(self, params):
```

Comment:
I'd suggest take `dd_keys` as one argument to init this class object.
It's quite wired that it uses an argument in the context.

Reply:
agree with hereischen,  self.dd_keys = dd_keys